### DISCUSSION TITLE
Google Season of Docs 2023

### DISCUSSION THREAD

**Discussion by pottekkat** [apache/apisix#8852] [Ideas]
**Update (2nd April 2023)**: Our proposal was not accepted by Google Season of Docs. However, all the proposed projects mentioned here will be worked on. If you are interested in contributing, feel free to do so. Thank you for all the ideas and enthusiasm. We will participate in more programs in the coming years. 

[Google Season of Docs](https://developers.google.com/season-of-docs) provides support for open source projects to improve their documentation and gives professional technical writers an opportunity to gain experience in open source.

Apache APISIX will be participating in this program this year. The [proposed projects](#1-move-out-documentation-from-the-others-section) are described below.

Please follow the steps below to show your interest:

1. Mention which project out of the [three described below](#1-move-out-documentation-from-the-others-section) you would like to work on in this discussion forum. Please indicate interest here before sending a statement of interest. Make sure you thoroughly go through each proposal and evaluate how you can solve it before moving on to the next step. We encourage you to [familiarize with Apache APISIX](#familiarizing-with-apache-apisix-and-docs) if you haven't already.
2. Send a [statement of interest](https://developers.google.com/season-of-docs/docs/tech-writer-statement) (please follow this template) describing (with examples) your previous technical writing experience showcasing the "skills required" indicated in each project. Choose only one project from the proposed projects below. You can send this to navendu@apache.org.

**Deadline to send the statement of interest: 3rd March 2023**

## Familiarizing with Apache APISIX and docs

1. You can start by running APISIX locally. All necessary documentation can be found on the [APISIX website](https://apisix.apache.org/).
2. Go through the existing documentation and identify the [problems mentioned in the proposal](#1-move-out-documentation-from-the-others-section).
4. Build and test the documentation locally. I encourage you to figure this out yourself. You can only build the apisix-website repo locally. But familiarize with how the docs for each of the APISIX projects are hosted in different repos. For example, this repo hosts the docs for APISIX.

**Please check the discussion to see if a question is already answered before asking a new one.**

If you have unanswered questions, please reach out in the discussion below or on our [Slack channel](https://apisix.apache.org/docs/general/join/#join-the-slack-channel).

### Problem

The "[Others](https://apisix.apache.org/docs/apisix/discovery/)" section contains unorganized documentation pages and subdirectories. It can be difficult for a user to intuitively navigate these pages.

### How would we measure success?
1. Reduction of pages in the "Others" section.
5. Decrease issues related to the discoverability of the documentation.
6. More categories/existing categories with a better ability to navigate.

### What skills would a technical writer need to work on this project?

**Must have**: Familiarity with Git and GitHub, familiarity with networks, and working with software like APISIX, markdown, and building sites locally with tools like Docusaurus.
**Nice to have**: Experience as an APISIX user, and understanding of SEO.

### Volunteers

@navendu-pottekkat, ?

### Problem

There have been [attempts](https://github.com/apache/apisix/issues/6461) to re-organize APISIX docs for some time now to improve intuitive navigation, but none has made any progress so far. Existing proposals are either stale or incomplete (do not address how to solve the problem). Lack of proper planning and organization of docs has resulted in users being unable to find the relevant docs and using our communication forums more.

### How would we measure success?

1. Complete proposal describing the expected state of the documentation and action plans to get there.
4. Decrease issues related to the discoverability of the documentation.
5. More categories/existing categories with a better ability to navigate.

### What skills would a technical writer need to work on this project?

**Must have**: Familiarity with Git and GitHub, familiarity with networks, and working with software like APISIX, markdown, building sites locally with tools like Docusaurus, and experience in working with large project documentation.
**Nice to have**: Experience as an APISIX user, and understanding of SEO.

### Volunteers

@navendu-pottekkat, ?

### Problem

APISIX has a lot of open issues across our forums and GitHub related to documentation. These need to be validated and fixed.

### How would we measure success?

1. Reduction of open issues tagged "documentation" or "docs" across our repos.

### What skills would a technical writer need to work on this project?

**Must have**: Familiarity with Git and GitHub, familiarity with networks, and working with software like APISIX, markdown, and building sites locally with tools like Docusaurus.
**Nice to have**: Experience as an APISIX user, and understanding of SEO.

### Volunteers

@navendu-pottekkat, ?

**Comment by Mich45** [comment]
Hi @navendu-pottekkat, I'd love to work on this project. How do you suggest I get started? Thanks!

**Reply by pottekkat** [reply]
You can start by:

1. Testing out the APISIX project.
2. Going through the current documentation.
3. Going through the proposed projects and selecting which project you have an interest in.
4. Identifying how you can contribute to the selected project.
5. Submitting a statement of interest to navendu@apache.org [as suggested by GSoD](https://developers.google.com/season-of-docs/docs/tech-writer-statement).

**Comment by Mich45** [comment]
Thanks @navendu-pottekkat . I'll do as you said.

**Comment by PaarthAgarwal** [comment]
Hi @navendu-pottekkat. The projects are fascinating, and I would love to work with Apache this season of docs! Setting up the codebase right now

**Comment by Prajwalprakash3722** [comment]
Hello there, hope you are doing great 🚀 @navendu-pottekkat; Apache is a organisation that has always inspired me and super interesting; would love to contribute in some way or the other; setting up my local codebase and going through docs now :)

**Comment by Amishakumari544** [comment]
Hi, @navendu-pottekkat I am experienced freelance technical writer and familiar with markdown, docasuarus, Git, and GitHub looking forward to working here.

**Comment by jasonbraganza** [comment]
Hi @navendu-pottekkat, I’m interested. Will do what you suggested [here](https://github.com/apache/apisix/discussions/8852#discussioncomment-4968784)

**Comment by aynp** [comment]
Hey,
I've some experience with APISIX and contributing to the Ingress Controller. I have the project set up and am also bit familiar with existing docs. I would love to be a part of it.

I decided to move forward with Project 1. Move out documentation from the "Others" section. I'll send my statement of interest soon.

**Comment by pottekkat** [comment]
@aynp @jasonbraganza @Amishakumari544 @Prajwalprakash3722 @PaarthAgarwal @Mich45 Please check the updated instructions: https://github.com/apache/apisix/discussions/8852#discussion-4856327

Thank you very much for showing interest! I'm really looking forward to seeing your proposals!

**Comment by money8203** [comment]
Hey @navendu-pottekkat I'd love to contribute to "Move out documentation from the "Others" section" or "Triage and fix open documentation issues"

**Comment by juzhiyuan** [comment]
Hi @navendu-pottekkat, for your records:

https://groups.google.com/g/season-of-docs-announce/c/fPi3f7rEbfY

**Reply by pottekkat** [reply]
Thanks @juzhiyuan we will submit our applications next month after we gather suggestions from our technical writers.

**Comment by Comfortade** [comment]
@navendu-pottekkat I will love to work on this fascinating project.

**Comment by siddhi-244** [comment]
I would like to work on the 2nd and 3rd project which is reorganizing  the docs and Triage and fix open documentation issues

**Reply by pottekkat** [reply]
You can send a statement of interest for any one of the proposed projects as described below.

**Comment by jasonbraganza** [comment]
@navendu-pottekkat I’d love to work on the third project. I’ll send my statement of interest by eod (India) today.

Doubt: when you say run locally, [installation via Docker](https://apisix.apache.org/docs/apisix/installation-guide/) on the desktop is a good approach and what you had in mind?

**Reply by pottekkat** [reply]
Yes. Docker is an easy way to run APISIX.

**Reply by yokwejuste** [reply]
@navendu-pottekkat how are we to send the statement? like directly to you or a particular email?

**Reply by jasonbraganza** [reply]
To navendu@apache.org

**Comment by Prajwalprakash3722** [comment]
Hey @navendu-pottekkat I would like to work on problem 1; will send my sop and interests by today/tomorrow eod; (IST) 

Thank you :)

**Comment by RajGM** [comment]
Hi @navendu-pottekkat 
I am interested in working on task 2. Reorganize documentation.
I have built relevant documentation in the past, some of them in personal projects: 

https://github.com/RajGM/GSoC_OrganizationsTracker#whats-included

https://github.com/RajGM/SecureMessaging#whats-included

Both include file structure in the readme.md file with almost the exact requirements for task 2.

https://hashnode.com/@rajgm

Currently, I am finishing my work at the LFX mentorship program.

**Comment by RakeshPotnuru** [comment]
Hey @navendu-pottekkat! I'm interested in participating in Season of Docs with Apache APISIX. sending the statement of interest 🚀.

**Comment by GhoulKingR** [comment]
Hey @navendu-pottekkat, I'm interested in working on task 3, "Triage and fix open documentation issues". I look forward to working on it. Thank you!

**Comment by rgrupesh** [comment]
@navendu-pottekkat I'm interested in task 1, which is about moving documentation out of the "Others" section. I'm really excited to work on this project! 
By the way, is there a deadline for submitting a statement of interest? I'm just wondering if it's better to submit early or if there's a specific deadline I should aim for. Thanks!

**Reply by pottekkat** [reply]
@rgrupesh You can send it by this Friday. It is better to submit early.

**Comment by pottekkat** [comment]
I have updated the deadline to send the statement of interest to **3rd March 2023**.

**Comment by dashohoxha** [comment]
I'd like to work on reorganizing documentation. I will send a statement of interest soon.

**Comment by FavourDaniel** [comment]
I would like to work on the reorganization of documentation.

**Comment by XFarooqi** [comment]
Hi @navendu-pottekkat 

I am interested and excited to work on Project 1 and Project 3.

Thanks

**Comment by pronoym99** [comment]
Hello @navendu-pottekkat 

Would love to participate in reorganising the documentation of Apisix. Would be sending my proposal in 2 - 3 days.

Thanks

**Reply by pottekkat** [reply]
Sorry, we are not accepting any new applications. The deadline, as mentioned in the discussion above, was on 3rd March. We have a lot of applications and it wouldn't be fair to the process.

**Reply by pronoym99** [reply]
Ok sure @navendu-pottekkat

**Reply by pottekkat** [reply]
@pronoym99 You can still contribute to the project. This program was only a means to encourage participation but all contributions are welcome.
